### PR TITLE
add partial kafka TLS (produce to one TLS kafka topic with new env vars)

### PR DESCRIPTION
**Pull request by gedge**
### What

enable one produced topic (`CSV_EXPORTED_PRODUCER_TOPIC`) to use TLS
(this topic sends to the XLSX exporter - being the only topic/consumer in our little TLS world at the moment!)

(other kafka producers/consumers continue with legacy/non-TLS)

depends on https://github.com/ONSdigital/dp-dataset-exporter-xlsx/pull/117

### How to review

ensure only the one producer has changed

requires env var changes (documented)

### Who can review

nice personages

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -10,8 +10,15 @@ import (
 // Config values for the application.
 type Config struct {
 	BindAddr                   string        `envconfig:"BIND_ADDR"`
+	KafkaLegacyAddr            []string      `envconfig:"LEGACY_KAFKA_ADDR"`
+	KafkaLegacyVersion         string        `envconfig:"LEGACY_KAFKA_VERSION"`
```
Comment (nshumoogum) [inline config/config.go:14]: Im not sure where you have used this - does this configuration need to be removed if it isn't used anymore?

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: config/config.go (hunk 1 above)
Comment (nshumoogum) [reply config/config.go:14]: It also makes me think there is a mistake in the codebase if TLS isn't set will this fail to consume/produce messages?

File: Makefile [hunk 2]
```diff
@@ -35,7 +35,7 @@ generate:
 
 .PHONY: audit
 audit:
-	go list -m all | nancy sleuth
+	go list -m all | nancy sleuth --exclude-vulnerability-file ./.nancy-ignore
```
Comment (nshumoogum) [inline Makefile]: I dont think you need the flag - nancy sleuth by default should ignore vulnerabilities in .nancy-ignore file

File: config/config.go (hunk 1 above)
Comment (gedge) [reply config/config.go:14]: the secrets will need to be updated, initially ("feature-flagged off") with these two as a copy of the existing values

File: Makefile (hunk 2 above)
Comment (gedge) [reply Makefile]: removed

**Review by nshumoogum — APPROVED** [review]

File: config/config.go (hunk 1 above)
Comment (nshumoogum) [reply config/config.go:14]: I just realised this config must be used in dp-kafka

**Pull request merged**